docs/source/notebooks/08_Getting_and_bias_correcting_CMIP6_data.ipynb

```diff
@@ -0,0 +1,1788 @@
+{
```

Comment:
<p>I suggest using a loop to avoid repeating the same code.</p><pre class="ql-syntax" spellcheck="false">out = []
for exp in ['historical', 'ssp585']:
    out[exp] = {}
    for variable in ['tasmin', 'tasmax', 'pr']:
        out[exp][variable] = ...
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/raven/pull/424/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
<p>This returns an error, and I am unsure of what exactly the code is doing. Basically putting all the data into a 2D array of exp and variable?</p><p> </p><p>This is the code I wrote according to the comment above:</p><p> </p><pre class="ql-syntax" spellcheck="false">out = []
&nbsp;&nbsp;for exp in ['historical','ssp585']:
&nbsp;&nbsp;&nbsp;&nbsp;out[exp] = {}
&nbsp;&nbsp;&nbsp;&nbsp;for variable in ['tasmin','tasmax','pr']:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;query = dict(experiment_id=exp, table_id='day', variable_id=variable, member_id='r1i1p1f1', source_id=climate_model)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;col_subset = col.search(require_all_on=['source_id'], **query)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mapper = fsCMIP.get_mapper(col_subset.df.zstore[0])
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;out[exp][variable] = historical_tasmin=subset.subset_shape(xr.open_zarr(mapper, consolidated=True).sel(time=slice(reference_start_day,reference_end_day)), basin_contour).mean({'lat','lon'}).reset_coords('height',drop=True)[variable].chunk(-1)
</pre><p> </p><p> </p><p>This is the error I get:</p><p> </p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [15], in &lt;cell line: 2&gt;()
      5 out = []
      6 for exp in ['historical','ssp585']:
----&gt; 7     out[exp] = {}
      8     for variable in ['tasmin','tasmax','pr']:
      9         query = dict(experiment_id=exp, table_id='day', variable_id=variable, member_id='r1i1p1f1', source_id=climate_model)

TypeError: list indices must be integers or slices, not str
</pre><p> </p><p> </p><p>I also tried changing the curly braces for brackets, same error:</p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [16], in &lt;cell line: 2&gt;()
      5 out = []
      6 for exp in ['historical','ssp585']:
----&gt; 7     out[exp] = []
      8     for variable in ['tasmin','tasmax','pr']:
      9         query = dict(experiment_id=exp, table_id='day', variable_id=variable, member_id='r1i1p1f1', source_id=climate_model)

TypeError: list indices must be integers or slices, not str
</pre><p> </p><p>Any insights?</p>

Reply:
<p>oups, start with <code>out={}</code></p>

Reply:
<p>Thanks, works now!</p>